servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1478,8 +1517,180 @@ public void setState(long fileId, SetStateOptions options)
     }
   }
 
-  private void setStateInternal(long fileId, long opTimeMs, SetStateOptions options)
+  /**
+   * Schedules a file for async persistence.
+   *
+   * @param fileId the id of the file for persistence
+   * @return the id of the worker that persistence is scheduled on
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public long scheduleAsyncPersistence(long fileId) throws FileDoesNotExistException {
+    // find the worker
+    long workerId = getWorkerStoringFile(fileId);
+
+    if (workerId == IdUtils.INVALID_WORKER_ID) {
+      LOG.warn("No worker found to schedule async persistence for file {}", fileId);
+      // no worker found, do nothing
+      return workerId;
+    }
+
+    // update the state
+    synchronized (mInodeTree) {
```

Comment:
Ok, so right now, there is absolutely no master fault tolerance for async persistence? I don't see how requesting async persistence will be remembered across a master crash.

If you plan to add it later, please add a TODO to the comments.

edit: I see. pollFilesToCheckpoint() adds the journal message, but there is no persistence between this call and the worker heartbeat. The "commit" point of requesting async persistence is actually when the master tells the worker, and not the request itself. That is intentional?